modules/irmaseal-pkg-server.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.irmaseal-pkg;
+in
+{
+
+  options.services.irmaseal-pkg = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        This option enables the IRMAseal-PKG server.
+      '';
+    };
+
+    package = mkOption {
+      type = types.path;
+      default = pkgs.irmaseal-pkg;
+      description = "The IRMAseal-PKG package to use";
+    };
+
+    host = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      description = ''
+        Host to bind this service to
+      '';
+    };
+
+    irma = mkOption {
+      type = types.str;
+      default = "https://irma-noauth.demo.sarif.nl";
+      description = ''
+        URL of the IRMA go server to use for authentication
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8087;
+      description = ''
+        TCP port to bind this service to
+      '';
+    };
+
+    publicKeyPath = mkOption {
+      type = types.path;
+      default = ./pkg.pub;
+      description = ''
+        Path to the public key
+      '';
+    };
+
+    secretKeyPath = mkOption {
+      type = types.path;
+      default = ./pkg.sec;
+      description = ''
+        Path to the private key
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.services.irmaseal-pkg = {
+      description = "IRMAseal PKG HTTP server";
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig.Restart = "always";
+      script = ''
```

Comment:
I don't think you need a script here. It's usually more common to use `serviceConfig.ExecStart` if it's just a one liner.

Reply:
Gotcha